libff/algebra/fields/binary_field.hpp

```diff
@@ -0,0 +1,101 @@
+/** @file
+ *****************************************************************************
+ Declaration of common API for all finite fields in the binary/ directory.
+
+ Currently NOT used by the fields in this library. This class is not actually
+ the parent class of any field. All APIs are enforced through tests instead.
+
+ The reason for this is to ensure high performance of all fields. This class
+ exists as documentation for common API between fields.
+
+ Includes fields F2^n for specified n. All of the binary entension fields must
+ implement all functions declared in this class.
+ *****************************************************************************
+ * @author     This file is part of libff, developed by SCIPR Lab
+ *             and contributors (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+#include <libff/algebra/field_utils/bigint.hpp>
+#include <vector>
+
+namespace libff {
+
+template<typename T>
+class BinaryField;
+
+/* The type parameter T is intended to be set to the child class
+ * when this class is extended. For example,
+ * class gf32 : public BinaryField<gf32> ...
+ */
+template<typename T>
+class BinaryField {
+public:
+    /* Functions unique to binary fields */
+
+    /** Returns the constituent bits in 64 bit words, in little-endian order. */
```

Comment:
Is there a reason why you use comments of the form `/* <my comment> */` and of the form `/** <my comment> */`?
If not, that's inconsistent and thus I'd favor one style everywhere.
If the goal is to use Doxygen, we need to stick to one supported code block style: https://www.doxygen.nl/manual/docblocks.html

Hence:
```
/**
 * <my comment>
 */
```
Would work best here

Reply:
/** */ is used for documentation, while /* */ is used to mark a new section of code, and // is general comments that explain some code, but is not high level enough to be included in documenation.

Reply:
The documentation `/** */` when it's one line and
```
/**
 *
 */
```
otherwise

Reply:
Oh right, didn't know `/**<comment> */` was also a valid Doxygen syntax. Sounds good!